core/domain/taskqueue_services_test.py

```diff
@@ -59,6 +59,32 @@ def test_exception_raised_when_email_task_params_is_not_serializable(self):
                 params,
                 0)
 
+    def test_defer_makes_the_correct_request(self):
+        correct_fn_identifier = '/task/deferredtaskshandler'
+        correct_url = feconf.TASK_URL_DEFERRED
+        correct_queue_name = taskqueue_services.QUEUE_NAME_EMAILS
+        correct_payload = {
+            'fn_identifier': '/task/deferredtaskshandler',
+            'args': [],
+            'kwargs': {}
+        }
+
+        swap_create_http_task = self.swap_with_checks(
+            taskqueue_services.platform_taskqueue_services,
+            'create_http_task',
+            lambda queue_name, url, payload=None, scheduled_for=None,
+                task_name=None: (
+                self.assertEqual(queue_name, correct_queue_name),
+                self.assertEqual(url, correct_url),
+                self.assertEqual(payload, correct_payload),
+                self.assertIsNone(scheduled_for),
+                self.assertIsNone(task_name)),
```

Comment:
Sorry if I wasn't clear; this wasn't what I meant. I meant something like this:

```python
self.swap_with_checks(
    subprocess, 'Popen', mock_popen,
    expected_args=[(['python'],), (['python2'],)],
    expected_kwargs=[{'shell': True}, {'shell': False}])
```

This example comes from our [backend tests wiki page](https://github.com/oppia/oppia/wiki/Backend-tests#backend-test-structure), which you should read through

Reply:
@U8NWXD Modified the test case accordingly. PTAL. Thanks!